### PR TITLE
Stops node_modules from being cached during CI builds

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
Addressing the recent build issues we've been having due to cache invalidation.

**Previous build times (range)**
* Azure: 12-14 mins
* Appveyor: 13-16 mins
* Travis: 10-12 mins
* Cicle: 12-14 mins

☝️ is given as ranges because I couldn't figure out how to export build history in order to do give better descriptive stats. Also all these times are from the past week.

**After this change**
* Azure: 11m44s
* Appveyor: 17m30s
* Travis: 11m41s
* Cicle: 12m34s

Since these times are within the ranges I saw in our build history for successful builds, removing the cache doesn't seem to have made a significant impact.

### CODE REVIEW

**Comment by shiftkey** [comment]
@iAmWillShepherd are you able to add into this PR the change in time between this build and a recent successful build on `development`? It'd be nice to know roughly how much this slows this down.

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,7 +32,6 @@ jobs:
             - .eslintcache
             - ~/.electron
             - ~/Library/Caches/Yarn/v1
```
Comment (shiftkey) [inline .circleci/config.yml]: I just had a this path on my new machine and I only have a `v4` key. Maybe this needs to be updated?

```
$ ls ~/Library/Caches/Yarn/
v4
```

Comment (iAmWillShepherd) [reply .circleci/config.yml]: yea, I think you're right since I got the same result.

**Comment by shiftkey** [comment]
> * Appveyor: 13-16 mins
> ...
> * Appveyor: 17m30s

This for me is the one I worry about - it's already the slowest build, and this PR makes it even slower compared to the others.

What if we left the change in #6799 in place for Appveyor only?

File: appveyor.yml [hunk 2]
```diff
@@ -8,7 +8,6 @@ environment:
 
 cache:
   - .eslintcache
-  - node_modules
   - '%USERPROFILE%\.electron'
   - '%LOCALAPPDATA%\Yarn'
```
Comment (shiftkey) [inline appveyor.yml]: Related to the previous ppoint about where `yarn` uses for a local cache:

```suggestion
   - '%LOCALAPPDATA%\Yarn\Cache\v4'
```

Comment (shiftkey) [reply appveyor.yml]: On my Win10 machine I see the `v2` folder has ~800MB of packages, so I think making this explicit will help speed up our Windows builds

**Comment by shiftkey** [comment]
@iAmWillShepherd I'm 👍 to getting this in and monitoring the build times over the `1.6.3` release process. Let me know if there's anything else you plan to do in this PR.

**Comment by iAmWillShepherd** [comment]
Nope, this should be good. 

I went ahead and remove the work-around I was using to get CI to pass in a971a98c94c6027107fef6917249f4e4056d33ad

**Comment by shiftkey** [comment]
Merging this in early to unblock some other CI work

**Pull request merged**